src/index.ts

```diff
@@ -67,41 +75,52 @@ export const compare = memoize((versionA: string | null, versionB: string | null
 		return 1;
 	}
 
-	versionA = normalize(versionA);
-	versionB = normalize(versionB);
+	const nVersionA = normalize(versionA);
+	const nVersionB = normalize(versionB);
 
-	const isAValid = semver.valid(versionA);
-	const isBValid = semver.valid(versionB);
+	const isAValid = semver.valid(nVersionA);
+	const isBValid = semver.valid(nVersionB);
 	if (isAValid && !isBValid) {
 		return 1;
 	}
 	if (!isAValid && isBValid) {
 		return -1;
 	}
 	if (!isAValid && !isBValid) {
-		if (versionA > versionB) {
+		if (nVersionA > nVersionB) {
 			return 1;
 		}
-		if (versionA < versionB) {
+		if (nVersionA < nVersionB) {
 			return -1;
 		}
 		return 0;
 	}
-	const semverResult = semver.compare(versionA, versionB);
+	const semverResult = semver.compare(nVersionA, nVersionB);
 	if (semverResult !== 0) {
 		return semverResult;
 	}
-	const revA = getRev(versionA);
-	const revB = getRev(versionB);
-	if (revA !== revB) {
-		return revA > revB ? 1 : -1;
-	}
-	const devA = Number(isDevelopmentVersion(versionA));
-	const devB = Number(isDevelopmentVersion(versionB));
-	if (devA !== devB) {
-		return devA > devB ? -1 : 1;
+
+	const suffixComparisonMethods = [
+		{ fn: getRev, reverse: false, normalize: true },
+		{ fn: isPreReleaseVariant, reverse: true, normalize: true },
+		{ fn: getPreReleaseVariant, reverse: false, normalize: true },
```

Comment:
So that pre-repeases come before the actual release but also , `-dev` comes before `-prod`.